azure_functions_worker/dispatcher.py

```diff
@@ -113,22 +113,29 @@ async def dispatch_forever(self):
             self._loop.set_task_factory(
                 lambda loop, coro: ContextEnabledTask(coro, loop=loop))
 
-            # Attach gRPC logging to the root logger
+            # Detach console logging before enabling GRPC channel logging
+            logger.info('Detaching console logging.')
+            disable_console_logging()
+
+            # Attach gRPC logging to the root logger. Since gRPC channel is
+            # established, should use it for system and user logs
             logging_handler = AsyncLoggingHandler()
             root_logger = logging.getLogger()
             root_logger.setLevel(logging.INFO)
             root_logger.addHandler(logging_handler)
-
-            # Since gRPC channel is established, should use it for logging
-            disable_console_logging()
-            logger.info('Detach console logging. Switch to gRPC logging')
+            logger.info('Switched to gRPC logging.')
+            logging_handler.flush()
 
             try:
                 await forever
             finally:
-                # Re-enable console logging when there's an exception
-                enable_console_logging()
+                logger.info('Detaching gRPC logging due to exception.')
+                logging_handler.flush()
                 root_logger.removeHandler(logging_handler)
+
+                # Reenable console logging when there's an exception
+                enable_console_logging()
+                logger.info('Switched to console logging due to exception.')
```

Comment:
Can this be a warning - we need to make it clear that console logging is not the standard anymore.